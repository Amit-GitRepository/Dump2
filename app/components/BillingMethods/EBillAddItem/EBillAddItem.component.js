/* EbillAddItem: show single service item and provide option to add ebill or email*/

import camelCase from 'lodash/camelCase';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './EBillAddItem.style';
import {Checkbox, ISText} from '../../Shared';
import {colors} from '../../../themes/constants.styles';
import {emailRegex} from '../../../config/ebill.config';
import {Keyboard, TextInput, View} from 'react-native';
import {translate} from '../../../language/i18n/helper';

class EBillAddItem extends Component {
  state = {
    valueMsisdn: '',
    valueEmail: '',
    errorMsisdn: null,
    errorEmail: null,
    isValidMsisdn: false,
    isValidEmail: false,
    editableMsisdn: true,
    editableEmail: true
  }

  componentWillMount () {
    const {accountType, defaultMSISDN} = this.props;
    if (['True Move H postpaid'].includes(accountType)) {
      this.setState({
        valueMsisdn: defaultMSISDN,
        editableMsisdn: false,
        isValidMsisdn: true
      });
    }
  }

  componentWillUpdate (nextProps) {
    if (nextProps.billingFormat !== this.props.billingFormat) {
      this.setState({errorMsisdn: null, errorEmail: null});
    }
  }

  onBillToggle = (event) => {
    this.props.onToggleSelection(this.props.accountId, event);
  }

  validValue = (value, type) => {
    if (type === 'EMAIL') {
      this.setState({
        valueEmail: value,
        isValidEmail: true,
        errorEmail: null
      });
    } else {
      this.setState({
        valueMsisdn: value,
        isValidMsisdn: true,
        errorMsisdn: null
      });
    }
  };

  invalidValue = (error, type) => {
    if (type === 'EMAIL') {
      this.setState({
        isValidEmail: false,
        errorEmail: error
      });
    } else {
      this.setState({
        isValidMsisdn: false,
        errorMsisdn: error
      });
    }
  };

  validateEmail = (value) => {
    if (value) {
      emailRegex.test(value)
        ? this.validValue(value, 'EMAIL')
        : this.invalidValue('BILLING_METHODS__ADD_INVALID_EMAIL', 'EMAIL');
    } else {
      this.invalidValue('BILLING_METHODS__ADD_REQUIRED_EMAIL', 'EMAIL');
    }
  };

  validateMSISDN = (value) => {
    const msisdnRegex = /^\d{10}$/;

    if (value) {
      msisdnRegex.test(value)
        ? this.validValue(value, 'SMS')
        : this.invalidValue('BILLING_METHODS__ADD_INVALID_MSISDN', 'SMS');
    } else {
      this.invalidValue('BILLING_METHODS__ADD_REQUIRED_MSISDN', 'SMS');
    }
  };

  onEndEditing = (event) => {
    const {billingFormat} = this.props;
    Keyboard.dismiss();
    if (billingFormat === 'EMAIL') {
      this.validateEmail(event.nativeEvent.text);
    } else {
      this.validateMSISDN(event.nativeEvent.text);
    }
  }

  onFocus = () => this.setState({
    errorMsisdn: null,
    isValidMsisdn: false,
    errorEmail: null,
    isValidEmail: false
  })

  onChangeText = (type) => (value) => {
    this.props.onValueUpdate(this.props.accountId, value);
    if (type === 'EMAIL') {
      this.setState({valueEmail: value});
    } else {
      this.setState({valueMsisdn: value});
    }
  };

  generateOption = (option) => {
    const {editableMsisdn, errorMsisdn, errorEmail, valueEmail, valueMsisdn} = this.state;
    let ebillOptionTitle =  translate('BILLING_METHODS__ADD_EMAIL');
    let errorOption  = errorEmail;
    let textViewOptions = {placeholder: translate('BILLING_METHODS__ADD_EMAIL_PLACEHOLDER'), editable: true, value: valueEmail};
    if (option === 'SMS') {
      ebillOptionTitle = translate('BILLING_METHODS__ADD_SMS');
      errorOption = errorMsisdn;
      textViewOptions = {placeholder: translate('BILLING_METHODS__ADD_SMS_PLACEHOLDER'), editable: editableMsisdn, keyboardType: 'numeric', maxLength: 10, onFocus: this.onFocus, value: valueMsisdn};
    }
    return (
      <View style={styles.ebillOptionContainer}>
        <View style={styles.ebillOptionInput}>
          <ISText type='SEMIBOLD' style={styles.ebillOption}>{ebillOptionTitle}</ISText>
          <TextInput {...textViewOptions} autoCorrect={false} onChangeText={this.onChangeText(option)} onEndEditing={this.onEndEditing}  placeholderTextColor={colors.PRIMARY_DISABLED_BG_TEXT} returnKeyType={'done'} style={styles.ebillOptionValue} underlineColorAndroid={colors.TRANSPARENT}
          />
        </View>
        {errorOption ? <ISText type='BOLD' style={styles.error}>{translate(errorOption)}</ISText> : null}
      </View>);
  }

  render () {
    const {billingFormat, productId, accountType, isSelected} = this.props;
    return (
      <View style={styles.container}>
        <View style={styles.serviceInfoContainer}>
          <View>
            <ISText type='SEMIBOLD' style={styles.serviceNumber}>{productId}</ISText>
            <ISText type='BOLD' style={styles[camelCase(accountType)]}>{accountType}</ISText>
          </View>
          <View>
            <Checkbox style={styles.checkbox} isChecked={isSelected} onChange={this.onBillToggle}/>
          </View>
        </View>
        {this.generateOption(billingFormat)}
      </View>
    );
  }
}

EBillAddItem.defaultProps = {
  defaultMSISDN: ''
};

EBillAddItem.propTypes = {
  accountType: PropTypes.string.isRequired,
  billingFormat: PropTypes.string.isRequired,
  productId: PropTypes.string.isRequired,
  defaultMSISDN: PropTypes.string,
  onToggleSelection: PropTypes.func.isRequired,
  onValueUpdate: PropTypes.func.isRequired,
  isSelected: PropTypes.bool.isRequired,
  accountId: PropTypes.string.isRequired
};

export default EBillAddItem;
