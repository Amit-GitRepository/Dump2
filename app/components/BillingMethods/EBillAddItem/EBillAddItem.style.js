import {colors, fonts} from '../../../themes/constants.styles';

export default {
  container: {
    padding: 15,
    flex: 1,
    borderTopWidth: 1,
    borderTopColor: colors.PRIMARY_BG_SEPARATOR
  },
  serviceInfoContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 5
  },
  serviceNumber: {
    fontSize: fonts.FONT_SIZE_MEDIUM
  },
  trueMoveHPostpaid: {
    color: colors.SECONDARY_TMOVE
  },
  trueMoveHPrepaid: {
    color: colors.SECONDARY_TMOVE
  },
  trueOnline: {
    color: colors.SECONDARY_TONLINE
  },
  trueVision: {
    color: colors.SECONDARY_TVISION
  },
  trueConvergence: {
    color: colors.PRIMARY_ACTIONABLE
  },
  checkbox: {
    marginTop: 10,
    marginRight: 5
  },
  ebillOptionContainer: {
    marginTop: 5
  },
  ebillOptionInput: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  ebillOption: {
    fontSize: fonts.FONT_SIZE_MEDIUM,
    marginRight: 70
  },
  ebillOptionValue: {
    fontFamily: fonts.FONT_FAMILY_LIGHT,
    fontSize: fonts.FONT_SIZE_NORMAL,
    flex: 1
  },
  error: {
    color: colors.PRIMARY_ACTIONABLE
  }
};
