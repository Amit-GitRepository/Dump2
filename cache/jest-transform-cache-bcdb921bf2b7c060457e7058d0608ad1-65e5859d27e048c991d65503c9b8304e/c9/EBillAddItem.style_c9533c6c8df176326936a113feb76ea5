Object.defineProperty(exports,"__esModule",{value:true});var cov_etfqk4qd7=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillingMethods/EBillAddItem/EBillAddItem.style.js',hash='6f7a7dd65a419a99397ce9177638b31987f771e4',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillingMethods/EBillAddItem/EBillAddItem.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../themes/constants.styles');exports.default=

{
container:{
padding:15,
flex:1,
borderTopWidth:1,
borderTopColor:_constants.colors.PRIMARY_BG_SEPARATOR},

serviceInfoContainer:{
flexDirection:'row',
justifyContent:'space-between',
marginBottom:5},

serviceNumber:{
fontSize:_constants.fonts.FONT_SIZE_MEDIUM},

trueMoveHPostpaid:{
color:_constants.colors.SECONDARY_TMOVE},

trueMoveHPrepaid:{
color:_constants.colors.SECONDARY_TMOVE},

trueOnline:{
color:_constants.colors.SECONDARY_TONLINE},

trueVision:{
color:_constants.colors.SECONDARY_TVISION},

checkbox:{
marginTop:10,
marginRight:5},

ebillOptionContainer:{
marginTop:5},

ebillOptionInput:{
flexDirection:'row',
alignItems:'center'},

ebillOption:{
fontSize:_constants.fonts.FONT_SIZE_MEDIUM,
marginRight:70},

ebillOptionValue:{
fontFamily:_constants.fonts.FONT_FAMILY_LIGHT,
fontSize:_constants.fonts.FONT_SIZE_NORMAL},

error:{
color:_constants.colors.PRIMARY_ACTIONABLE}};