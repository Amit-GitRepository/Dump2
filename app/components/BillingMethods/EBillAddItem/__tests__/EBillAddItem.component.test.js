import EBillAddItem from '../EBillAddItem.component';
import React from 'react';
import renderer from 'react-test-renderer'; // Enzyme adapter for React 16

describe('EBillAddItem component', () => {
  it('renders correctly with SMS option', () => {
    const component = renderer.create(<EBillAddItem
      accountType='True Online'
      productId='1234567'
      billingFormat='SMS'
    />).toJSON();
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with EMAIL option', () => {
    const component = renderer.create(<EBillAddItem
      accountType='True Online'
      productId='1234567'
      billingFormat='EMAIL'
    />).toJSON();
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with SMS and default msisdn option', () => {
    const component = renderer.create(<EBillAddItem
      accountType='TrueMove H Postpaid'
      productId='1234567890'
      defaultMSISDN='1234567890'
      billingFormat='SMS'
    />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
