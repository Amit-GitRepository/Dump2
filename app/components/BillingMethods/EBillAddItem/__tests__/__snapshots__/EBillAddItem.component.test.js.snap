// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EBillAddItem component renders correctly with EMAIL option 1`] = `
<View
  style={
    Object {
      "borderTopColor": "#E9E9EA",
      "borderTopWidth": 1,
      "flex": 1,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 5,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#303C46",
            "fontFamily": "SukhumvitSet-SemiBold",
            "fontSize": 18,
          }
        }
      >
        1234567
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#009FE3",
            "fontFamily": "SukhumvitSet-Bold",
          }
        }
      >
        True Online
      </Text>
    </View>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "marginRight": 5,
              "marginTop": 10,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "#F93E60",
                "borderRadius": 5,
                "borderWidth": 2,
                "padding": 2,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "color": "#B9C1C9",
                  },
                  Object {
                    "fontFamily": "iservice",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              юдл
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#303C46",
            "fontFamily": "SukhumvitSet-SemiBold",
            "fontSize": 18,
            "marginRight": 70,
          }
        }
      >
        Email
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        editable={true}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="Enter email address"
        placeholderTextColor="#B9C1C9"
        returnKeyType="done"
        style={
          Object {
            "flex": 1,
            "fontFamily": "SukhumvitSet-Light",
            "fontSize": 16,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;

exports[`EBillAddItem component renders correctly with SMS and default msisdn option 1`] = `
<View
  style={
    Object {
      "borderTopColor": "#E9E9EA",
      "borderTopWidth": 1,
      "flex": 1,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 5,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#303C46",
            "fontFamily": "SukhumvitSet-SemiBold",
            "fontSize": 18,
          }
        }
      >
        1234567890
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#FBBC1E",
            "fontFamily": "SukhumvitSet-Bold",
          }
        }
      >
        TrueMove H Postpaid
      </Text>
    </View>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "marginRight": 5,
              "marginTop": 10,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "#F93E60",
                "borderRadius": 5,
                "borderWidth": 2,
                "padding": 2,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "color": "#B9C1C9",
                  },
                  Object {
                    "fontFamily": "iservice",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              юдл
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#303C46",
            "fontFamily": "SukhumvitSet-SemiBold",
            "fontSize": 18,
            "marginRight": 70,
          }
        }
      >
        SMS
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        editable={true}
        keyboardType="numeric"
        maxLength={10}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        onFocus={[Function]}
        placeholder="Enter mobile number"
        placeholderTextColor="#B9C1C9"
        returnKeyType="done"
        style={
          Object {
            "flex": 1,
            "fontFamily": "SukhumvitSet-Light",
            "fontSize": 16,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;

exports[`EBillAddItem component renders correctly with SMS option 1`] = `
<View
  style={
    Object {
      "borderTopColor": "#E9E9EA",
      "borderTopWidth": 1,
      "flex": 1,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 5,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#303C46",
            "fontFamily": "SukhumvitSet-SemiBold",
            "fontSize": 18,
          }
        }
      >
        1234567
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#009FE3",
            "fontFamily": "SukhumvitSet-Bold",
          }
        }
      >
        True Online
      </Text>
    </View>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "marginRight": 5,
              "marginTop": 10,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "#F93E60",
                "borderRadius": 5,
                "borderWidth": 2,
                "padding": 2,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "color": "#B9C1C9",
                  },
                  Object {
                    "fontFamily": "iservice",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              юдл
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#303C46",
            "fontFamily": "SukhumvitSet-SemiBold",
            "fontSize": 18,
            "marginRight": 70,
          }
        }
      >
        SMS
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        editable={true}
        keyboardType="numeric"
        maxLength={10}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        onFocus={[Function]}
        placeholder="Enter mobile number"
        placeholderTextColor="#B9C1C9"
        returnKeyType="done"
        style={
          Object {
            "flex": 1,
            "fontFamily": "SukhumvitSet-Light",
            "fontSize": 16,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;
